server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -446,6 +453,13 @@ protected final void failIfNotIndexed() {
         }
     }
 
+    protected final void failIfNotIndexedNorDocValues() {
```

Comment:
Maybe `failIfNotSearchable`?

Reply:
I've renamed this (because of expensive queries check), but prefer the more concrete name.

Reply:
Fair enough, and I like how the two checks are combined, very neat.